internal/locality/locality_test.go

```diff
@@ -138,105 +163,260 @@ var _ = Describe("Localities", func() {
+					Expect(localities[1].ID).To(Equal("storage-51"))
+					Expect(localities[2].Class).To(Equal(fdbv1beta2.ProcessClassLog))
+					Expect(localities[2].ID).To(Equal("log-1"))
+					Expect(localities[3].Class).To(Equal(fdbv1beta2.ProcessClassTransaction))
+					Expect(localities[3].ID).To(Equal("tlog-1"))
+				})
 			})
 		})
 
-		When("when the storage class is preferred over transaction class", func() {
+		When("a multi-region setup is used", func() {
+			var primaryID, remoteID, primarySatelliteID, remoteSatelliteID string
+
 			BeforeEach(func() {
-				cluster.Spec.CoordinatorSelection = []fdbv1beta2.CoordinatorSelectionSetting{
-					{
-						ProcessClass: fdbv1beta2.ProcessClassStorage,
-						Priority:     10,
-					},
-					{
-						ProcessClass: fdbv1beta2.ProcessClassTransaction,
-						Priority:     1,
+				// Generate random names to make sure we test different alphabetical orderings.
+				primaryID = internal.GenerateRandomString(10)
+				remoteID = internal.GenerateRandomString(10)
+				primarySatelliteID = internal.GenerateRandomString(10)
+				remoteSatelliteID = internal.GenerateRandomString(10)
+
+				localities = make([]Info, 12)
+				idx := 0
+				// Make sure the result is randomized
+				dcIDs := []string{primaryID, primarySatelliteID, remoteID, remoteSatelliteID}
+				rand.Shuffle(len(dcIDs), func(i, j int) {
+					dcIDs[i], dcIDs[j] = dcIDs[j], dcIDs[i]
+				})
+
+				for _, dcID := range dcIDs {
+					for i := 0; i < 3; i++ {
+						pClass := fdbv1beta2.ProcessClassStorage
+						// The first locality will be a log process
+						if i == 0 {
+							pClass = fdbv1beta2.ProcessClassLog
+						}
+
+						localities[idx] = Info{
+							ID: dcID + strconv.Itoa(i),
+							LocalityData: map[string]string{
+								fdbv1beta2.FDBLocalityZoneIDKey: dcID + "-z" + strconv.Itoa(i),
+								fdbv1beta2.FDBLocalityDCIDKey:   dcID,
+							},
+							Class: pClass,
+						}
+
+						idx++
+					}
+				}
+
+				// Randomize the order of localities.
+				rand.Shuffle(len(localities), func(i, j int) {
+					localities[i], localities[j] = localities[j], localities[i]
+				})
+
+				cluster.Spec.DatabaseConfiguration = fdbv1beta2.DatabaseConfiguration{
+					UsableRegions: 2,
+					Regions: []fdbv1beta2.Region{
+						{
+							DataCenters: []fdbv1beta2.DataCenter{
+								{
+									ID:       primaryID,
+									Priority: 1,
+								},
+								{
+									ID:        primarySatelliteID,
+									Priority:  1,
+									Satellite: 1,
+								},
+								{
+									ID:        remoteSatelliteID,
+									Priority:  0,
+									Satellite: 1,
+								},
+							},
+						},
+						{
+							DataCenters: []fdbv1beta2.DataCenter{
+								{
+									ID:       remoteID,
+									Priority: 0,
+								},
+								{
+									ID:        remoteSatelliteID,
+									Priority:  1,
+									Satellite: 1,
+								},
+								{
+									ID:        primarySatelliteID,
+									Priority:  0,
```

Comment:
why is primary lower than remote here? or does the priority not actually signify primary vs remote? (though I don't see a distinction made in the tests so I don't think it matters)

Reply:
This is the config for the remote side and the remote side should prefer the remote satellite to store mutations and not the primary satellite. See: https://github.com/apple/foundationdb/wiki/Multi-Region-Replication#concepts